Pause and purge message delivery (#20284)
cloudflare/cloudflare-docs: src/content/docs/queues/configuration/pause-purge.mdx

---
title: Pause and purge Queues
pcx_content_type: concept
sidebar:
  order: 2
---
import { WranglerConfig, Type, MetaInfo } from "~/components";

## Pause Delivery

> You can pause delivery of messages from your Queue to any connected consumers. Pausing a queue is useful to manage downtime (such as if your consumer Worker is unhealthy), without losing any messages.

Reviewer: ```suggestion
You can pause delivery of messages from your Queue to any connected consumers. Pausing a Queue is useful when managing downtime (such as if your consumer Worker is unhealthy) without losing any messages.
```

Queues continue to receive and store messages even while delivery is paused. Messages in a paused queue are still subject to expiry, if the messages become older than the queue message retention period.

> Pausing affects both [push-based consumer Workers](/queues/reference/how-queues-works#consumers), and [pull based consumers](/queues/configuration/pull-consumers).

Reviewer: ```suggestion
Pausing affects both [push-based consumer Workers](/queues/reference/how-queues-works#consumers) and [pull based consumers](/queues/configuration/pull-consumers).
```

> ### Pause and resume delivery using Wrangler

Reviewer: ```suggestion
### Pause and resume delivery using Wrangler

```

> The following command will pause message delivery from your queue:

Reviewer: ```suggestion
The following command will pause message delivery from your queue:

```

```sh
$ npx run wrangler queues pause-delivery <QUEUE-NAME>
```
- `queue-name` <Type text="string" /> <MetaInfo text="required" />
  - The name of the queue which delivery should be paused.

The following command will resume message delivery:
```sh
$ npx run wrangler queues resume-delivery <QUEUE-NAME>
```

> - `queue-name` <Type text="string" /> <MetaInfo text="required" />
>   - The name of the queue from which delivery should be resumed.

Reviewer: ```suggestion

- `queue-name` <Type text="string" /> <MetaInfo text="required" />
  - The name of the queue for which delivery should be resumed.
```

### What happens to HTTP Pull consumers with a paused queue?
When a queue is paused, messages cannot be pulled by an [HTTP pull based consumer](/queues/configuration/pull-consumers). Requests to pull messages will receive a `409` response, along with an error message stating `queue_delivery_paused`.

## Purge queue
Purging a queue permanently deletes any messages currently stored in the Queue. Purging is useful while developing a new application, especially to clear out any test data. It can also be useful in production to handle scenarios when a batch of bad messages have been sent to a Queue.

Note that any in flight messages, which are currently being processed by consumers, might still be processed. Messages sent to a queue during a purge operation might not be purged. Any delayed messages will also be deleted from the Queue.

> :::note

Reviewer: ```suggestion
:::caution
```

Purging a queue is an irreversible operation. Make sure to use this operation carefully.
:::

> ### Purge queue using wrangler

Reviewer: ```suggestion
### Purge queue using Wrangler
```
